localstack-core/localstack/utils/container_utils/container_client.py

```diff
@@ -1193,35 +1241,12 @@ def parse_additional_flags(
         if args.env_files:
             env_vars = env_vars if env_vars is not None else {}
             for env_file in args.env_files:
-                try:
-                    with open(env_file, mode="rt") as f:
-                        env_file_lines = f.readlines()
-                except FileNotFoundError as e:
-                    LOG.error(
-                        "Specified env file '%s' not found. Please make sure the file is properly mounted into the LocalStack container. Error: %s",
-                        env_file,
-                        e,
-                    )
-                    raise
-                except OSError as e:
-                    LOG.error(
-                        "Could not read env file '%s'. Please make sure the LocalStack container has the permissions to read it. Error: %s",
-                        env_file,
-                        e,
-                    )
-                    raise
-                for idx, line in enumerate(env_file_lines):
-                    line = line.strip()
-                    if not line or line.startswith("#"):
-                        # skip comments or empty lines
-                        continue
-                    lhs, separator, rhs = line.partition("=")
-                    if rhs or separator:
-                        env_vars[lhs] = rhs
-                    else:
-                        # No "=" in the line, only the name => lookup in local env
-                        if env_value := os.environ.get(lhs):
-                            env_vars[lhs] = env_value
+                env_vars.update(Util._read_docker_cli_env_file(env_file))
+
+        if args.compose_env_files:
+            env_vars = env_vars if env_vars is not None else {}
+            for env_file in args.compose_env_files:
+                env_vars.update(dotenv.dotenv_values(env_file))
```

Comment:
We should maybe document the precedence order somewhere. Basically that the --compose-env-file arguments are evaluated after the --env-file arguments and might override them

Reply:
Agreed! I will try to make it part of the docs for the lambda docker flags!